### PR TITLE
Add reframe tests for spack concretisation and module file generation

### PR DESCRIPTION
**Pull request by craigmeyer**
Adds a set of reframe tests designed to be run during deployment of the software stack to check spack concretisation and generation/functionality of module files. A summary of key changes:

- A script to install reframe, `install_reframe.sh`, separate to other packages. This is to ensure the reframe for the stack being installed is the one used for the tests. This is executed immediately after `install_python.sh` in the the `install_software_stack.sh` script
- A new directory, `systems/setonix/rfm_files` which holds the files needed for the Reframe tests.
- Two new scripts, `run_rfm_concretisation_tests.sh` and `run_rfm_module_tests.sh`. These scripts run the tests themselves. The first is called right after `concretise_environments.sh` while the second is run at the end of `post_installation_operations.sh`

The only modifications to scripts which should be required are:

- Setting `INSTALL_PREFIX`, `INSTALL_GROUP`, and `SYSTEM` as usual in the job submission script
- Setting `RFM_STORAGE_DIR` in the `settings.sh` file to the directory where one wishes to store the reframe logs and stage/output files. I've set it to `INSTALL_PREFIX/rfm_results` here, but if you wish it to be stored somewhere else, edit this.
- Setting `reframe_version` in the `settings.sh` to the version relevant to the stack being deployed.

If running the tests manually (i.e. not as part of `install_software_stack.sh`) it is important to run them from the repo used for the deployment of the stack you are testing. Otherwise, some of the script and test logic will break.

### CODE REVIEW

File: scripts/install_reframe.sh
Comment (ddeeptimahanti) [inline scripts/install_reframe.sh:1]: Reframe is currently installed as part of utils env. Do we need this script again?

File: scripts/post_installation_operations.sh
Comment (ddeeptimahanti) [inline scripts/post_installation_operations.sh:1]: I can't see run_all_rfm_module_tests.sh script in the files

File: scripts/install_reframe.sh
Comment (ddeeptimahanti) [reply scripts/install_reframe.sh:1]: Ok, we need reframe to run reframe concretization script

File: scripts/post_installation_operations.sh
Comment (craigmeyer) [reply scripts/post_installation_operations.sh:1]: Yes, an oversight on my part. That was an old name for the script that is now the `run_rfm_module_tests.sh` script. I will quickly amend that.

**Review by ddeeptimahanti — APPROVED** [review]

**Pull request merged**